src/include/webdriverxx/conversions.h

```diff
@@ -9,286 +9,291 @@
+			WEBDRIVERXX_CHECK(it != map.end(), detail::Fmt() << "No \"" << name << "\" in JsonObject");
+			return FromJson<T>(it->second);
+		}
+
+		template<typename T>
+		inline T GetOptional(const std::string& name, const T& default_value = T()) const {
+			const auto& map = value_.get<picojson::object>();
+			const auto it = map.find(name);
+			return it != map.end() ? FromJson<T>(it->second) : default_value;
+		}
+
+		template<typename T>
+		inline JsonObject& Set(const std::string& name, const T& value) {
+			value_.get<picojson::object>()[name] = ToJson(value);
+			return *this;
+		}
+
+		inline
+			bool Has(const std::string& name) const {
+			const auto& map = value_.get<picojson::object>();
+			return map.find(name) != map.end();
+		}
+
+		operator const picojson::value& () const {
+			return value_;
+		}
+
+	private:
+		picojson::value value_;
+	};
+
+	namespace conversions_detail {
+
+		struct DefaultTag {};
+		struct IterableTag {};
+
+		using namespace detail;
+
+		template<typename T>
+		struct Tag :
+			if_<is_iterable<T>, type_is<IterableTag>,
+			type_is<DefaultTag>
+			> {};
+
+		template<typename T>
+		inline picojson::value ToJsonImpl(const T& value, DefaultTag) {
+			// Compile error here usually indicates
+			// that compiler doesn't know how to convert the type T
+			// to the picojson::value. Define CustomToJson
+			// function (see examples below) in the T's namespace
+			// to resolve the issue.
+			return picojson::value(value);
+		}
+
+		template<typename T>
+		inline picojson::value ToJsonImpl(const T& value, IterableTag) {
+			typedef typename std::iterator_traits<decltype(std::begin(value))>::value_type Item;
+			picojson::value result = picojson::value(picojson::array());
+			picojson::array& dst = result.get<picojson::array>();
+			std::transform(std::begin(value), std::end(value), std::back_inserter(dst), [](const Item& item) {
+				return ToJson(item);
+				});
+			return result;
+		}
+
+	} // conversions_detail
+
+	inline
+		picojson::value CustomToJson(const char* value) {
+		return picojson::value(value);
+	}
+
+	inline
+		picojson::value CustomToJson(const std::string& value) {
+		return ToJson(value.c_str());
+	}
+
+	inline
+		picojson::value CustomToJson(const picojson::value& value) {
+		return value;
+	}
+
+	inline
+		picojson::value CustomToJson(const picojson::object& value) {
+		return picojson::value(value);
+	}
+
+	inline
+		picojson::value CustomToJson(const JsonObject& value) {
+		return static_cast<picojson::value>(value);
+	}
 
-template<typename T>
-inline T FromJson(const picojson::value& value);
+	inline
+		picojson::value CustomToJson(int value) {
+		return picojson::value(static_cast<double>(value));
+	}
 
-class JsonObject { // copyable
-public:
-	JsonObject() : value_(picojson::object()) {}
-	explicit JsonObject(const picojson::object& object) : value_(object) {}
+	inline
+		picojson::value CustomToJson(uint64_t value) {
+		return picojson::value(static_cast<double>(value));
```

Comment:
this part may a better way is to conver uint64 to string...however, I haven't had time to try yet

Reply:
@No-47, Opposite, need to remove uint64_t and static_cast and rewrite it into double.

```
namespace picojson {

  class value {
    ...

  public:

    typedef std::vector<value> array;
    typedef std::map<std::string, value> object;

    value();                               // create a null object
    explicit value(bool b);                // create a boolean object
    explicit value(double n);              // create a number object
    explicit value(const std::string& s);  // create a string object
    explicit value(const array& a);        // create an array object
    explicit value(const object& o);       // create an "object"

    bool is<picojson::null>() const;       // check if the object is "null"
```

More info in README.md https://github.com/kazuho/picojson